compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/generators/DependencyGraphFirGenerator.kt

```diff
@@ -339,9 +350,16 @@ internal class DependencyGraphFirGenerator(session: FirSession) :
                   name = valueParameterSymbol.name,
                   key = Keys.RegularParameter,
                   typeProvider = {
```

Comment:
The use of `typeProvider` here seems pointless since you can't any type parameters to the constructor. `typeProvider` isn't actually that lazy and will run before the `createConstructor` function exits. It's used when you need to add value parameters / return type to depend on type parameters of a _**function**_. So the whole `withArguments` mapping is completely useless.

Same goes for generating functions for both the value parameters and return type. Both values should just be provided directly and don't need the `typeProvider` style.

Reply:
> So the whole withArguments mapping is completely useless.

Gotcha! Makes sense